### PR TITLE
Add more unit tests

### PR DESCRIPTION
**Pull request by ppamorim**
In this PR:

 - Added most of the required unit tests with the correct float verification and its required mock files.
 - Corrected massive mistakes done in the data mapping. (non critical ones).
 - Added missing `stop_limit_sell_order` and `test_stop_limit_sell_order`. #hotfix

### CODE REVIEW

File: src/account.rs [hunk 1]
```diff
@@ -472,7 +472,63 @@ impl Account {
         };
         let order = self.build_order(sell);
         let request = build_signed_request(order, self.recv_window)?;
-        self.client.post_signed::<()>(API::Spot(Spot::OrderTest), request)
+        self.client.post_signed::<Void>(API::Spot(Spot::OrderTest), request)
+    }
+
+    /// Place a stop limit buy order
```
Comment (dorak88783) [inline src/account.rs:478]: sell order

File: tests/account_tests.rs [hunk 2]
```diff
@@ -1,18 +1,424 @@
...
+
+        let mock_get_account = mock("GET", "/api/v3/account")
+            .with_header("content-type", "application/json;charset=UTF-8")
+            .match_query(Matcher::Regex("recvWindow=1234&timestamp=\\d+&signature=.*".into()))
+            .with_body_from_file("tests/mocks/account/get_account.json")
+            .create();
+
+        let config = Config::default()
+            .set_rest_api_endpoint(mockito::server_url())
+            .set_recv_window(1234);
```
Comment (dorak88783) [inline tests/account_tests.rs:63]: I put this `recv_window=1234` in my first test case, maybe it's not needed in all of them.

**Comment by dorak88783** [comment]
Very very nice, impressive amount of work.

**Review by dorak88783 — APPROVED** [review]

File: tests/account_tests.rs (hunk 2 above)
Comment (ppamorim) [reply tests/account_tests.rs:63]: I think all signed methods require it, no?

**Comment by ppamorim** [comment]
@dorak88783 I am fighting with the compiler, `cargo build` in my machine does not compile the `doctest` by some reason...

**Comment by dorak88783** [comment]
> @dorak88783 I am fighting with the compiler, `cargo build` in my machine does not compile the `doctest` by some reason...

you should be able to see exactly what github is doing by looking at the config in `.github/action`; if you run exactly the same on your machine then there's some other configuration option. Do you have the latest version of rust?

I never worked with `doctest` myself.

**Comment by dorak88783** [comment]
@ppamorim  I'm only not really sold on the change from `()` to the custom-made `Void`. I thought `Void` was a built-in but it's something custom. I don't have enough Rust experience to really judge this.

**Comment by ppamorim** [comment]
> @ppamorim  I'm only not really sold on the change from `()` to the custom-made `Void`. I thought `Void` was a built-in but it's something custom. I don't have enough Rust experience to really judge this.

`()` causes serde to crash.

**Comment by ppamorim** [comment]
> > @dorak88783 I am fighting with the compiler, `cargo build` in my machine does not compile the `doctest` by some reason...
> 
> 
> 
> you should be able to see exactly what github is doing by looking at the config in `.github/action`; if you run exactly the same on your machine then there's some other configuration option. Do you have the latest version of rust?
> 
> 
> 
> I never worked with `doctest` myself.

It's the same strangely or not. Anyway it's fixed.

**Comment by wisespace-io** [comment]
@ppamorim Great work here.

**Comment by ppamorim** [comment]
> @ppamorim Great work here.

Thank you very very much!

Would you like to rename `Void` to something else? I am not sure how to name it properly.

**Comment by wisespace-io** [comment]
> > @ppamorim Great work here.
> 
> Thank you very very much!
> 
> Would you like to rename `Void` to something else? I am not sure how to name it properly.

Is it just an empty struct? You may use a synonym (adjective) such as Emptied, Blank, Unfilled

**Comment by dorak88783** [comment]
@ppamorim I now see there's also a create called `void`, maybe this also provides a more uniform/standard way to use this Void? I didn't check it myself yet. https://docs.rs/void/1.0.2/void

**Comment by ppamorim** [comment]
@wisespace-io @dorak88783 I think there is a way to map `Void` (internally it will be called `Empty` to `()` which makes more sense and it won't break the code.

**Comment by ppamorim** [comment]
`.map(|empty| ())` did the job.

**Pull request merged**